src/lib/libdylink.js

```diff
@@ -904,7 +904,9 @@ var LibraryDylink = {
     // We need to set rpath in flags based on the current library's rpath.
     // We can't mutate flags or else if a depends on b and c and b depends on d,
     // then c will be loaded with b's rpath instead of a's.
-    flags = {...flags, rpath: { parentLibPath: libName, paths: metadata.runtimePaths }}
+    var dso = LDSO.loadedLibsByName[libName];
```

Comment:
If the module is already loaded (i.e. already in loadedLibsByName) why/how would end up in this function which is loading it?

Reply:
The module is not fully loaded yet here. The `exports` section would still have 'loading' value in it.

The call graph is:

- `loadDynamicLibrary`
  - adds `dso` field in `LSDO.loadedLibsByName` with exports field not filled out.
  - calls `loadWebAssemblyModule`
     - `loadWebAssemblyModule`
       - [the codepointer in this comment] locates the library from the file system
       - loads the WASM module
       - return with exports field inside the module
   - fills out the exports field in the dso object (the module is fully loaded now).
   - return

So... I don't think `dso` would be the best place to put this information, but I wasn't sure what other object to use.